packages/buckets/src/buckets.spec.ts

```diff
@@ -242,4 +253,97 @@ describe('Buckets...', () => {
+      await bobBuckets.getToken(bob)
+      const { root, threadID } = await aliceBuckets.getOrCreate('createbuck')
+      aliceThread = threadID
+
+      // scope bob's client to the same thread
+      bobBuckets.withThread(aliceThread)
+
+      if (!root) throw Error('bucket creation failed')
+      rootKey = root.key
+
+      // Push private file for only alice
+      let stream = fs.createReadStream(path.join(pth, 'file1.jpg'))
+      await aliceBuckets.pushPath(rootKey, privatePath, stream)
+
+      // Push file to be shared with bob
+      stream = fs.createReadStream(path.join(pth, 'file2.jpg'))
+      await aliceBuckets.pushPath(rootKey, sharedFile, stream)
+    })
+
+    it('grant peer2 write access', async function () {
+      if (isBrowser) return this.skip()
+      if (!bobBuckets || !aliceBuckets || !aliceThread || !rootKey) throw Error('setup failed')
+
+      // Update access roles to grant bob access
+      const roles = new Map()
+      roles.set(bobPubKey, 2)
+      // Grants access at 'path/to'
+      await aliceBuckets.pushPathAccessRoles(rootKey, sharedPath, roles)
+
+      // Check that our pulled permissions are as expected
+      const shared = await aliceBuckets.pullPathAccessRoles(rootKey, sharedPath)
+      expect(shared.get(bobPubKey)).to.equal(2)
+    })
+
+    it('overwrite an existing shared file', async function () {
+      if (isBrowser) return this.skip()
+      if (!bobBuckets || !aliceBuckets || !aliceThread || !rootKey) throw Error('setup failed')
+
+      // Test that bob sees the same permissions, including himself
+      const perms = await bobBuckets.pullPathAccessRoles(rootKey, sharedPath)
+      expect(perms.get(bobPubKey)).to.equal(2)
+
+      // Over-write the file in the shared path
+      const stream = fs.createReadStream(path.join(pth, 'file2.jpg'))
+      // Pushing to an existing shared file works: sharedFile = 'path/to/file2.jpg'
+      await bobBuckets.pushPath(rootKey, sharedFile, stream)
+    })
+
+    it('remove a file in shared path', async function () {
+      if (isBrowser) return this.skip()
+      if (!bobBuckets || !aliceBuckets || !aliceThread || !rootKey) throw Error('setup failed')
+
+      try {
+        await bobBuckets.removePath(rootKey, sharedFile)
+        throw wrongError
+      } catch (err) {
+        expect(err).to.not.equal(wrongError)
+        expect(err.message).to.equal('app denied net record body: permission denied')
+      }
+    })
+
+    it('add a new file into a shared path should fail', async function () {
+      if (isBrowser) return this.skip()
+      if (!bobBuckets || !aliceBuckets || !aliceThread || !rootKey) throw Error('setup failed')
+
+      try {
+        const stream = fs.createReadStream(path.join(pth, 'file2.jpg'))
+        await bobBuckets.pushPath(rootKey, 'path/to/bobby.jpg', stream)
+        throw wrongError
+      } catch (err) {
+        expect(err).to.not.equal(wrongError)
+        expect(err.message).to.equal('2')
```

Comment:
Is it possible we're just not passing the full error message along from pushPath?

Reply:
Yeh that was the case:
https://github.com/textileio/js-hub/blob/8834c8ffee622ca4d4ab0fc6c8ca072c56249b3d/packages/buckets/src/api/index.ts#L426. We're swallowing the error message and only passing along the error code there.

Reply:
What we really want is something more like this:

```
onEnd: async (status: grpc.Code, message: string, _trailers: grpc.Metadata) => {
    if (status !== grpc.Code.OK) {
      reject(new Error(message))
    }
  resolve()
},
```

Reply:
We should actually just include that fix this in PR to save ourselves the future headache, and to fix that test.

Reply:
done